src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx

```diff
@@ -0,0 +1,380 @@
+  padding-bottom: 0.5rem;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`
+
+const Value = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  font-size: 1rem;
+  line-height: 1.5rem;
+  color: ${colorsV3.gray900};
+  text-transform: capitalize;
+`
+
+const LoadingDotsWrapper = styled.div`
+  width: 100%;
+  display: flex;
+  justify-content: center;
+`
+
+const ErrorMessage = styled(motion.div)`
+  background-color: ${colorsV3.purple500};
+  border-radius: 8px;
+  padding: 1.25rem;
+  margin-bottom: 10px;
+  color: ${colorsV3.gray900};
+`
+
+const DateInputModalWrapper = styled.div<{
+  isOpen: boolean
+}>`
+  width: 100%;
+  height: 100%;
+  position: absolute;
+  background: ${hexToRgba(colorsV3.white, 0.8)};
+  top: 0;
+  left: 0;
+  transition: all 0.2s;
+  visibility: ${(props) => (props.isOpen ? 'visible' : 'hidden')};
+  opacity: ${(props) => (props.isOpen ? 1 : 0)};
+  padding: 1rem;
+  display: flex;
+  align-items: center;
+  border-radius: 8px;
+`
+
+const StyledDateInput = styled(DateInputForm)<{
+  modal: boolean
+}>`
+  ${(props) =>
+    props.modal &&
+    `
+  top: 50%;
+  transform: translateY(-50%);
+  `}
+`
+
+const getDefaultDateValue = (quote: OfferQuote) => {
+  if (quote.startDate) {
+    return parse(quote.startDate, 'yyyy-MM-dd', new Date())
+  }
+
+  if (hasCurrentInsurer(quote)) {
+    return null
+  }
+
+  return new Date()
+}
+
+const getDateFormat = match<MarketLabel, string>([
+  ['SE', 'dd MMM yyyy'],
+  ['NO', 'dd/MM/yyyy'],
+  ['DK', 'yyyy-MM-dd'],
+])
+
+const DateForm: React.FC<{
+  quoteCartId: string
+  quote: OfferQuote
+  offerData: OfferData
+  isSingleStartDateBundle?: boolean
+  isSplit: boolean
+  setShowError: (showError: boolean) => void
+  modal?: boolean
+  size: Size
+}> = ({
+  quoteCartId,
+  quote,
+  offerData,
+  isSingleStartDateBundle = false,
+  isSplit,
+  setShowError,
+  modal = false,
+  size,
+}) => {
+  const textKeys = useTextKeys()
+  const { isoLocale, marketLabel } = useCurrentLocale()
+
+  const [dateValue, setDateValue] = useState(() => getDefaultDateValue(quote))
+  const [dateLocale, setDateLocale] = useState<Locale | null>(null)
+  const [isLoadingPickedStartDate, setIsLoadingPickedStartDate] = useState(
+    false,
+  )
+  const [datePickerOpen, setDatePickerOpen] = useState(false)
+
+  const [setStartDate] = useSetStartDateMutation()
+
+  useEffect(() => {
+    setDateValue(getDefaultDateValue(quote))
+  }, [quote])
+
+  useEffect(() => {
+    getLocaleImport(isoLocale).then((m) => setDateLocale(m.default))
+  }, [isoLocale])
+
+  const getDateLabel = () => {
+    if (dateValue) {
+      if (isToday(dateValue)) {
+        return textKeys.SIDEBAR_STARTDATE_CELL_VALUE_NEW()
+      }
+
+      return format(dateValue, getDateFormat(marketLabel)!, {
+        locale: dateLocale!,
+      })
+    }
+  }
+
+  const handleSetDate = async (newDateValue: Date | null) => {
```

Comment:
This is a copy of `StartDate` component placed inside `Offer`. The major difference is this method which uses the new API.